### ISSUE TITLE
ModuleNotFoundError: No module named 'vectordb'

### ISSUE THREAD

**Issue by TheRManVT** [wAIfu-DEV/w-AI-fu_v2#47]
**Describe the bug**
After installing and starting the program, an error for "vectordb", then the splash screen appears, and after it goes away the program loads until you either close the CMD window or until you have turned into bones.

**To Reproduce**
1. Install via INSTALL.bat
2. Wait for it to install
3. Start via shortcut "Run w-AI-fu"
4. Splash screen and "vectordb" error appear, then after 2-3 minutes the splash screen vanishes but the program doesn't start
5. (optional) Wait till the heat death of the universe

**Expected behavior**
w-AI-fu GUI shows up

**Screenshots**
![image](https://github.com/wAIfu-DEV/w-AI-fu_v2/assets/97435041/25594d7d-72bb-4d62-9541-b0032a4961d6)

**Additional context**
I have Python 3.10.11, Node.js 19.8.1 installed, upgraded to the latest pip version and installed version 2.0.6
![image](https://github.com/wAIfu-DEV/w-AI-fu_v2/assets/97435041/db7f47d4-65af-4fd2-8814-f644004292dd)

**Comment by moo7md1** [comment]
I was having this problem and the solution was to go back to Python 3.10.11 and update pip to the latest

**Comment by NorbsCode** [comment]
> I was having this problem and the solution was to go back to Python 3.10.11 and update pip to the latest

I'm having the same issues. Tried with 3.10.11 and latest pip and not working :(

**Comment by TheRManVT** [comment]
> I was having this problem and the solution was to go back to Python 3.10.11 and update pip to the latest

I've already installed Python 3.10.11 and upgraded to the latest pip version before I installed version 2.0.6 but it still happened.
(Also, I left the window running for 8 hours and the GUI did not show up)

**Comment by wAIfu-DEV** [comment]
Could you try going in the `source/app/vectordb` folder and remove the `venv` folder? It should create a new one after restart

**Comment by TheRManVT** [comment]
> Could you try going in the `source/app/vectordb` folder and remove the `venv` folder? It should create a new one after restart

I've done that but it still happens.
After starting, the error and splash screen appear, and then after that the GUI still doesn't show up...

**Comment by DumbatorCZ** [comment]
It is because "vectordb2==0.1.9" stated in the requirements wants TF-text 2.13, but that one is no longer build for windows. Your options are switching to WSL or build TF-text from source.

**Comment by wAIfu-DEV** [comment]
I doubt that's the reason since some windows machines can run the program just fine

**Comment by Salty4Daddy** [comment]
What I have told you there worked for me. :)
Also running in Py311

**Comment by Hisagi10** [comment]
i run it like @ Salty4Daddy  said but i get 1 more error and solve it , maybe there is better way am not expert here but  if some one dont now hot to install the files and want the fix this is the way :

1- run CDM 
2- go to the folder that have venv on it like this :  "\w-AI-fu_v2-main\source\app\vectordb\"  just copy the location from the folder 
3- Run " venv\Scripts\activate" 
4- Run "pip install tensorflow-intel==2.13.0" 
5- Run "pip install vectordb2==0.1.9 --no-deps" 
here try to run the app , in my case i get this error  " ModuleNotFoundError: No module named 'sentence_transformers'" if you get the same error as me go to step 6
6- run  "pip install sentence-transformers" 
and done this is make the AI work it just will donwload some files before it start 

note : i get this error after step 6 : "ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
vectordb2 0.1.9 requires tensorflow-text, which is not installed."
which idk why it happen because i already install them in step 4,5 but i ignore it and run the program and it work fine. 

again not expert so i just list what i did to run the program

**Comment by SaturnFiveLive** [comment]
Thank you for that, I did get further in launching but I am stuck at faiss
![image](https://github.com/wAIfu-DEV/w-AI-fu_v2/assets/157905810/c3d6a1db-882f-4e0f-8602-d11789f87bf3)

**Comment by SaturnFiveLive** [comment]
I got a little bit further following above.  The GUI asked me to install pip install git+https://github.com/vioshyvo/mrpt/ which I did, but now it hangs on "creating electron window..."

Tired uninstalling mrpt it made no difference.
![image](https://github.com/wAIfu-DEV/w-AI-fu_v2/assets/157905810/cb3e8dbf-465f-4930-be20-92cb421f1c09)
![image](https://github.com/wAIfu-DEV/w-AI-fu_v2/assets/157905810/0fa736e8-4c15-4b66-958a-601242f0561b)

**Comment by gg5150** [comment]
I’m having the same problem here…

**Comment by gg5150** [comment]
Can someone please help? I’m no good at coding, never done it before tbh so idk what to do

**Comment by Hisagi10** [comment]
> Can someone please help? I’m no good at coding, never done it before tbh so idk what to do

have you check my method to fix the problem ? i write it step by step , if you dont no anything about coding or there is some steps not clear , tell me and i will explain it more

**Comment by gg5150** [comment]
> > Can someone please help? I’m no good at coding, never done it before tbh so idk what to do
> 
> have you check my method to fix the problem ? i write it step by step , if you dont no anything about coding or there is some steps not clear , tell me and i will explain it more

I appreciate it! Thank you. Um, well, ‘CDM’ is the command prompt, right? Or no? What I did was I went to that folder (w-AI-fu-v2-main/source/app/vectordb) and replaced that name with CMD in the box, opening the command prompt inside the folder. Then I tried to open the venv/scripte/activate but from the folder. Should I have done it from the command prompt? After that I attempted to install everything you said in the vectordb folder from the CMD, but if anything was supposed to show up in there, it didn’t. Also, the ‘no module named sentence-transformers” error didn’t come up for me when I attempted to run the app, just the regular problem I was having. But I did get the error about the ‘pip dependency resolver’ in the CMD.

**Comment by Blackwolfaus** [comment]
I'm just reporting the same issues as above
![Screenshot 2024-01-30 220805](https://github.com/wAIfu-DEV/w-AI-fu_v2/assets/132946981/34e90e9a-87cf-4c78-bc28-a4ac324bca05)

**Comment by wAIfu-DEV** [comment]
I'll be pushing an update in the week to at least get more debug infos during the installation of the dependencies, won't fix it but should give us an idea of what's going wrong

**Comment by Hisagi10** [comment]
> Reference



> > > Can someone please help? I’m no good at coding, never done it before tbh so idk what to do
> > 
> > 
> > have you check my method to fix the problem ? i write it step by step , if you dont no anything about coding or there is some steps not clear , tell me and i will explain it more
> 
> I appreciate it! Thank you. Um, well, ‘CDM’ is the command prompt, right? Or no? What I did was I went to that folder (w-AI-fu-v2-main/source/app/vectordb) and replaced that name with CMD in the box, opening the command prompt inside the folder. Then I tried to open the venv/scripte/activate but from the folder. Should I have done it from the command prompt? After that I attempted to install everything you said in the vectordb folder from the CMD, but if anything was supposed to show up in there, it didn’t. Also, the ‘no module named sentence-transformers” error didn’t come up for me when I attempted to run the app, just the regular problem I was having. But I did get the error about the ‘pip dependency resolver’ in the CMD.

yeah my bad i meant CMD , am not really sure what you did so i will just walk you more step and i hope it help 

Run CMD 
 the defauil path will be in C Drive ( your windows Driver ) you need to change it to where  " Venv " folder exist  
 
 if the Wa-Ai-Fu files on another driver lets say its in  "D" 
1- Run " D: " inside CMD 
the CMD path will change to "D Drive "
 2- now go to "w-AI-fu_v2-main\source\app\vectordb "  and copy the path
 3- go to CMD Run " CD (past the location here )" 
 now it will change location to vectordb folder inside Wa-Ai-fu files 
 4- run " venv\Scripts\activate" in CMD 
 
 to make sure it work correctly you will find the path something  like this " (venv)  D:\XXX\XXX \w-AI-fu_v2-main\source\app\vectordb> "   you will see (venv) at the start this mean it work 
 
 now run 
5- Run "pip install tensorflow-intel==2.13.0" In CMD
6- Run "pip install vectordb2==0.1.9 --no-deps" in CMD 
Dont Close The CMD 
run Wa-Ai-Fu  if it work good , if not and you get  "No module named 'sentence_transformers" error 
7- run "pip install sentence-transformers" in CMD 
 
i hope this work now if its still not clear let me now i will try to explain it more : )

**Comment by wAIfu-DEV** [comment]
I managed to replicate the error by uninstalling Python and installing Python 3.11.7, but still works for me on Python 3.10.X

**Comment by wAIfu-DEV** [comment]
Also there seems to be an issue where the dependencies will not be installed correctly if you don't have this installed: [https://visualstudio.microsoft.com/visual-cpp-build-tools/](https://visualstudio.microsoft.com/visual-cpp-build-tools/)

**Comment by wAIfu-DEV** [comment]
And you'll need to install this
![image](https://github.com/wAIfu-DEV/w-AI-fu_v2/assets/132222556/4bae1add-a96c-4e4a-909f-69bfb4c01b0f)

**Comment by TheRManVT** [comment]
Thanks DEV.
It now works
![image](https://github.com/wAIfu-DEV/w-AI-fu_v2/assets/97435041/e5a9517e-933b-48c5-94dd-e9175e668528)
